packages/plugin-lightningcss/src/plugin.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  getBrowserslistWithDefault,
+  mergeChainedOptions,
+  parseMinifyOptions,
+} from '@rsbuild/shared';
+import type {
+  ModifyBundlerChainUtils,
+  RsbuildPlugin,
+  BundlerChain,
+} from '@rsbuild/shared';
+import browserslist from '@rsbuild/shared/browserslist';
+import { browserslistToTargets } from 'lightningcss';
+import path from 'node:path';
+import type {
+  LightningCssLoaderOptions,
+  PluginLightningcssOptions,
+} from './types';
+
+const PLUGIN_NAME = 'rsbuild:lightningcss';
+
+const applyLightningcssLoader = ({
+  chain,
+  utils: { CHAIN_ID },
+  browserslist: browserslistUserConfig,
+  options,
+}: {
+  chain: BundlerChain;
+  utils: ModifyBundlerChainUtils;
+  browserslist: string[];
+  options: PluginLightningcssOptions | undefined;
+}) => {
+  const defaultOptions = {
+    targets: browserslistToTargets(browserslist(browserslistUserConfig)),
+  } satisfies LightningCssLoaderOptions;
+
+  const mergedOptions: LightningCssLoaderOptions = mergeChainedOptions({
+    defaults: defaultOptions,
+    options: options?.transform,
+  });
+
+  [CHAIN_ID.RULE.CSS, CHAIN_ID.RULE.SASS, CHAIN_ID.RULE.LESS].forEach(
+    (ruleId) => {
+      const rule = chain.module.rule(ruleId);
+      const use = rule.use(CHAIN_ID.USE.LIGHTNINGCSS);
+
+      use.loader(path.resolve(__dirname, './loader')).options(mergedOptions);
+
+      switch (ruleId) {
+        case CHAIN_ID.RULE.SASS:
+          use.before(CHAIN_ID.USE.RESOLVE_URL);
+          break;
+        case CHAIN_ID.RULE.LESS:
+          use.before(CHAIN_ID.USE.LESS);
+          break;
+        case CHAIN_ID.RULE.CSS:
+          use.after(CHAIN_ID.USE.CSS);
+          break;
+      }
+      rule.uses.delete(CHAIN_ID.USE.POSTCSS);
+    },
+  );
+};
+
+const applyLightningCssMinifyPlugin = async ({
+  chain,
+  utils: { CHAIN_ID },
+  browserslist: browserslistUserConfig,
+  options,
+}: {
+  chain: BundlerChain;
+  browserslist: string[];
+  utils: ModifyBundlerChainUtils;
+  options: PluginLightningcssOptions | undefined;
+}) => {
+  const defaultOptions = {
+    targets: browserslistToTargets(browserslist(browserslistUserConfig)),
```

Comment:
`browserslistToTargets(browserslist(browserslistUserConfig))` is called twice, we can reuse the result for both loader and minimizer.